.github/workflows/failure_free_docs.yml

```diff
@@ -0,0 +1,29 @@
+name: failure_free_docs
+on:
+  pull_request:
+    paths:
+      - "docs/fail/**"
+
+permissions:
+  contents: read
+
+jobs:
+  checks_docs_for_failures:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
```

Comment:
I don't think an explicit Git ref is needed (eg the check-changelog check doesn't use one)

```suggestion
```